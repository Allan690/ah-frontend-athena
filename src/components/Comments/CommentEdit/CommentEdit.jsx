import React from "react";
import PropTypes from "prop-types";
import { MDBIcon } from "mdbreact";
import "./CommentEdit.scss";

const CommentEdit = ({ clickReply, clickDelete, clickEdit, id}) => {

  return (
    <div className="reply container row">
      <div className="col-8">
        <span id='reply'
          onClick={() => {
            clickReply(id);
          }}
          className="text-primary"
        >
          <MDBIcon icon="reply" />
          reply&nbsp;&nbsp;
        </span>
        <span id="edit"
          onClick={() => {
            clickEdit(id);
          }}
          className="text-primary"
        >
          <MDBIcon icon="edit" />
          Edit&nbsp;&nbsp;
        </span>
        <span id='delete'
          onClick={() => {
            clickDelete(id);
          }}
          className="text-primary"
        >
          <MDBIcon icon="trash" />
          delete&nbsp;&nbsp;
        </span>
      </div>
    </div>
  );
};

CommentEdit.propTypes = {
  clickReply: PropTypes.func,
  clickDelete: PropTypes.func,
  clickEdit: PropTypes.func,
  id: PropTypes.number
};

CommentEdit.defaultProps = {
  clickReply: () => {},
  clickDelete: () => {},
  clickEdit: () => {},
  id: 0
};
export default CommentEdit;
