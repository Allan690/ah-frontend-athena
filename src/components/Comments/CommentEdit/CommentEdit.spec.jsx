import { shallow, mount } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import CommentEdit from "./CommentEdit";

const props = {
  clickReply: jest.fn(),
  clickDelete: jest.fn(),
  clickEdit: jest.fn(),
  onClick: jest.fn(),
};
const onClick = jest.fn()

let store;
const mockStore = configureStore();
store = mockStore({comment: {}});
const mockCommentEdit = jest.fn();
describe("CommentEdit test", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CommentEdit {...props} onClick={onClick} />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render a container", () => {
    expect(wrapper.find("span")).toHaveLength(3);
  });

  it("should simulate clickReply event2223", () => {
    const wrapper2 = mount(
      <Provider store={store}>
        <CommentEdit {...props} onClick={onClick} />
      </Provider>)
      wrapper2.find('span').first().simulate('click')
      expect(props.clickReply).toBeCalled()
      
  });

  it("test handle clickDelete event", () => {
    wrapper = shallow(<CommentEdit clickDelete={mockCommentEdit} />);
    wrapper.find('#delete').simulate('click')
    expect(props.clickReply).toBeCalled()
  });

  it("test handle clickedit event", () => {
    wrapper = shallow(<CommentEdit />);
    wrapper.find('#edit').simulate('click')
    expect(props.clickReply).toBeCalled()
  });
});
