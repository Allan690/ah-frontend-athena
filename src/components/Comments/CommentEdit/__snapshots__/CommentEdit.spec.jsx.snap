// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentEdit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentEdit
    clickDelete={[MockFunction]}
    clickEdit={[MockFunction]}
    clickReply={[MockFunction]}
    id={0}
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-8"
      >
        <span
          className="text-primary"
          id="reply"
          onClick={[Function]}
        >
          <t
            border={false}
            brand={false}
            className=""
            fab={false}
            fal={false}
            far={false}
            fixed={false}
            flip=""
            icon="reply"
            inverse=""
            light={false}
            list={false}
            pull=""
            pulse={false}
            regular={false}
            rotate=""
            size=""
            spin={false}
            stack=""
          />
          reply  
        </span>
        <span
          className="text-primary"
          id="edit"
          onClick={[Function]}
        >
          <t
            border={false}
            brand={false}
            className=""
            fab={false}
            fal={false}
            far={false}
            fixed={false}
            flip=""
            icon="edit"
            inverse=""
            light={false}
            list={false}
            pull=""
            pulse={false}
            regular={false}
            rotate=""
            size=""
            spin={false}
            stack=""
          />
          Edit  
        </span>
        <span
          className="text-primary"
          id="delete"
          onClick={[Function]}
        >
          <t
            border={false}
            brand={false}
            className=""
            fab={false}
            fal={false}
            far={false}
            fixed={false}
            flip=""
            icon="trash"
            inverse=""
            light={false}
            list={false}
            pull=""
            pulse={false}
            regular={false}
            rotate=""
            size=""
            spin={false}
            stack=""
          />
          delete  
        </span>
      </div>,
      "className": "reply container row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="text-primary"
            id="reply"
            onClick={[Function]}
          >
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="reply"
              inverse=""
              light={false}
              list={false}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size=""
              spin={false}
              stack=""
            />
            reply  
          </span>,
          <span
            className="text-primary"
            id="edit"
            onClick={[Function]}
          >
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="edit"
              inverse=""
              light={false}
              list={false}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size=""
              spin={false}
              stack=""
            />
            Edit  
          </span>,
          <span
            className="text-primary"
            id="delete"
            onClick={[Function]}
          >
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="trash"
              inverse=""
              light={false}
              list={false}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size=""
              spin={false}
              stack=""
            />
            delete  
          </span>,
        ],
        "className": "col-8",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="reply"
                inverse=""
                light={false}
                list={false}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size=""
                spin={false}
                stack=""
              />,
              "reply  ",
            ],
            "className": "text-primary",
            "id": "reply",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "border": false,
                "brand": false,
                "className": "",
                "fab": false,
                "fal": false,
                "far": false,
                "fixed": false,
                "flip": "",
                "icon": "reply",
                "inverse": "",
                "light": false,
                "list": false,
                "pull": "",
                "pulse": false,
                "regular": false,
                "rotate": "",
                "size": "",
                "spin": false,
                "stack": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "reply  ",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="edit"
                inverse=""
                light={false}
                list={false}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size=""
                spin={false}
                stack=""
              />,
              "Edit  ",
            ],
            "className": "text-primary",
            "id": "edit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "border": false,
                "brand": false,
                "className": "",
                "fab": false,
                "fal": false,
                "far": false,
                "fixed": false,
                "flip": "",
                "icon": "edit",
                "inverse": "",
                "light": false,
                "list": false,
                "pull": "",
                "pulse": false,
                "regular": false,
                "rotate": "",
                "size": "",
                "spin": false,
                "stack": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Edit  ",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="trash"
                inverse=""
                light={false}
                list={false}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size=""
                spin={false}
                stack=""
              />,
              "delete  ",
            ],
            "className": "text-primary",
            "id": "delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "border": false,
                "brand": false,
                "className": "",
                "fab": false,
                "fal": false,
                "far": false,
                "fixed": false,
                "flip": "",
                "icon": "trash",
                "inverse": "",
                "light": false,
                "list": false,
                "pull": "",
                "pulse": false,
                "regular": false,
                "rotate": "",
                "size": "",
                "spin": false,
                "stack": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "delete  ",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-8"
        >
          <span
            className="text-primary"
            id="reply"
            onClick={[Function]}
          >
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="reply"
              inverse=""
              light={false}
              list={false}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size=""
              spin={false}
              stack=""
            />
            reply  
          </span>
          <span
            className="text-primary"
            id="edit"
            onClick={[Function]}
          >
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="edit"
              inverse=""
              light={false}
              list={false}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size=""
              spin={false}
              stack=""
            />
            Edit  
          </span>
          <span
            className="text-primary"
            id="delete"
            onClick={[Function]}
          >
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="trash"
              inverse=""
              light={false}
              list={false}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size=""
              spin={false}
              stack=""
            />
            delete  
          </span>
        </div>,
        "className": "reply container row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="text-primary"
              id="reply"
              onClick={[Function]}
            >
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="reply"
                inverse=""
                light={false}
                list={false}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size=""
                spin={false}
                stack=""
              />
              reply  
            </span>,
            <span
              className="text-primary"
              id="edit"
              onClick={[Function]}
            >
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="edit"
                inverse=""
                light={false}
                list={false}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size=""
                spin={false}
                stack=""
              />
              Edit  
            </span>,
            <span
              className="text-primary"
              id="delete"
              onClick={[Function]}
            >
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="trash"
                inverse=""
                light={false}
                list={false}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size=""
                spin={false}
                stack=""
              />
              delete  
            </span>,
          ],
          "className": "col-8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <t
                  border={false}
                  brand={false}
                  className=""
                  fab={false}
                  fal={false}
                  far={false}
                  fixed={false}
                  flip=""
                  icon="reply"
                  inverse=""
                  light={false}
                  list={false}
                  pull=""
                  pulse={false}
                  regular={false}
                  rotate=""
                  size=""
                  spin={false}
                  stack=""
                />,
                "reply  ",
              ],
              "className": "text-primary",
              "id": "reply",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "border": false,
                  "brand": false,
                  "className": "",
                  "fab": false,
                  "fal": false,
                  "far": false,
                  "fixed": false,
                  "flip": "",
                  "icon": "reply",
                  "inverse": "",
                  "light": false,
                  "list": false,
                  "pull": "",
                  "pulse": false,
                  "regular": false,
                  "rotate": "",
                  "size": "",
                  "spin": false,
                  "stack": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "reply  ",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <t
                  border={false}
                  brand={false}
                  className=""
                  fab={false}
                  fal={false}
                  far={false}
                  fixed={false}
                  flip=""
                  icon="edit"
                  inverse=""
                  light={false}
                  list={false}
                  pull=""
                  pulse={false}
                  regular={false}
                  rotate=""
                  size=""
                  spin={false}
                  stack=""
                />,
                "Edit  ",
              ],
              "className": "text-primary",
              "id": "edit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "border": false,
                  "brand": false,
                  "className": "",
                  "fab": false,
                  "fal": false,
                  "far": false,
                  "fixed": false,
                  "flip": "",
                  "icon": "edit",
                  "inverse": "",
                  "light": false,
                  "list": false,
                  "pull": "",
                  "pulse": false,
                  "regular": false,
                  "rotate": "",
                  "size": "",
                  "spin": false,
                  "stack": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "Edit  ",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <t
                  border={false}
                  brand={false}
                  className=""
                  fab={false}
                  fal={false}
                  far={false}
                  fixed={false}
                  flip=""
                  icon="trash"
                  inverse=""
                  light={false}
                  list={false}
                  pull=""
                  pulse={false}
                  regular={false}
                  rotate=""
                  size=""
                  spin={false}
                  stack=""
                />,
                "delete  ",
              ],
              "className": "text-primary",
              "id": "delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "border": false,
                  "brand": false,
                  "className": "",
                  "fab": false,
                  "fal": false,
                  "far": false,
                  "fixed": false,
                  "flip": "",
                  "icon": "trash",
                  "inverse": "",
                  "light": false,
                  "list": false,
                  "pull": "",
                  "pulse": false,
                  "regular": false,
                  "rotate": "",
                  "size": "",
                  "spin": false,
                  "stack": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "delete  ",
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
